src/vm/docs/mod.rs

```diff
@@ -264,6 +264,46 @@ Note: This function is only available starting with Stacks 2.1.",
 "#,
 };
 
+const PARSE_PRINCIPAL_API: SimpleFunctionAPI = SimpleFunctionAPI {
+    name: None,
+    signature: "(parse-principal property-name principal-address)",
+    description: "A principal value is a concatenation of two things: a single *version byte*
+(indicating the type of account and the type of network that this principal can spend tokens on),
+and a 20-byte *public key hash* (indicating the principal's unique identity).
+`parse-principal` is used to decompose a principal into its component parts.
+
+By setting `property-name` to `version`, one recovers version byte.
+
+By setting `property-name` to `pub-key-hash`, one recovers the 20-byte buffer representing the 
+public key hash of the user.
+
+Note: This function is only available starting with Stacks 2.1.",
+    example: r#"
+(parse-principal version 'STB44HYPYAT2BB2QE513NSP81HTMYWBJP02HPGK6) ;; Returns u26
+(parse-principal pub-key-hash 'STB44HYPYAT2BB2QE513NSP81HTMYWBJP02HPGK6) ;; Returns 0x164247d6f2b425ac5771423ae6c80c754f7172b0
+"#,
+};
+
+const ASSEMBLE_PRINCIPAL_API: SimpleFunctionAPI = SimpleFunctionAPI {
+    name: None,
+    signature: "(assemble-principal version-byte public-key-hash)",
+    description: "A principal value is a concatenation of two things: a single *version byte*
+(indicating the type of account and the type of network that this principal can spend tokens on),
+and a 20-byte *public key hash* (characterizing the principal's unique identity).
+`assemble-principal` takes such a *version byte*, of type `uint`, 
+and a *public key hash*, of type `(buff 20)`, and creates a corresponding object of type `principal`.
+
+The version byte should be `22` for a single-signature account on mainnet, `20`
+for a multi-signature account on mainnet, `26` for a single-signature account on
+a testnet, `21` for a multi-signature account on a testnet. The public key hash
+should be a 20-byte buffer containing the hash of a public key.
```

Comment:
Can you also give the hexadecimal representation (0x..) for this values? Developers are manipulating bytes.